aten/src/ATen/record_function.cpp

```diff
@@ -252,10 +252,12 @@ class CallbackManager {
       bool is_start) {
     try {
       if (is_start) {
-        ctx = rfcb.start()(rf);
+        ctx = rfcb.start() ? rfcb.start()(rf) : nullptr;
       }
       else {
-        rfcb.end()(rf, ctx.get());
+        if (rfcb.end()) {
+          rfcb.end()(rf, ctx.get());
+        }
```

Comment:
I there a need to cache the return value from `rfcb.end()` to prevent a race condition where the value may change in between or will that not happen by construction?